### ISSUE TITLE
torch.distributions.Gamma.rsample() no longer works

### ISSUE THREAD

**Issue by fritzo** [pytorch/pytorch#4319]
@alicanb observed that `_standard_gamma_grad()` is no longer working. This broken test is due to skipped tests in #4260 
```
pytest -v test/test_distributions.py
```
<details>

```
=================================================== FAILURES ====================================================
___________________________________ TestDistributions.test_gamma_sample_grad ____________________________________

self = <test_distributions.TestDistributions testMethod=test_gamma_sample_grad>

    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
    def test_gamma_sample_grad(self):
        self._set_rng_seed(1)
        num_samples = 100
        for alpha in [1e-3, 1e-2, 1e-1, 1e0, 1e1, 1e2, 1e3, 1e4]:
            alphas = Variable(torch.Tensor([alpha] * num_samples), requires_grad=True)
            betas = Variable(torch.ones(num_samples))
            x = Gamma(alphas, betas).rsample()
>           x.sum().backward()

test/test_distributions.py:415:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
torch/autograd/variable.py:103: in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

variables = (Variable containing:
1.00000e-07 *
  3.8609
[torch.DoubleTensor of size 1]
,)
grad_variables = (Variable containing:
 1
[torch.DoubleTensor of size 1]
,), retain_graph = False
create_graph = False

    def backward(variables, grad_variables=None, retain_graph=None, create_graph=False):
        """Computes the sum of gradients of given variables w.r.t. graph leaves.

        The graph is differentiated using the chain rule. If any of ``variables``
        are non-scalar (i.e. their data has more than one element) and require
        gradient, the function additionally requires specifying ``grad_variables``.
        It should be a sequence of matching length, that contains gradient of
        the differentiated function w.r.t. corresponding variables (``None`` is an
        acceptable value for all variables that don't need gradient tensors).

        This function accumulates gradients in the leaves - you might need to zero
        them before calling it.

        Arguments:
            variables (sequence of Variable): Variables of which the derivative will be
                computed.
            grad_variables (sequence of (Tensor, Variable or None)): Gradients w.r.t.
                each element of corresponding variables.  Any tensors will be
                automatically converted to Variables that are volatile unless
                ``create_graph`` is ``True``.  None values can be specified for scalar
                Variables or ones that don't require grad. If a None value would
                be acceptable for all grad_variables, then this argument is optional.
            retain_graph (bool, optional): If ``False``, the graph used to compute the grad
                will be freed. Note that in nearly all cases setting this option to ``True``
                is not needed and often can be worked around in a much more efficient
                way. Defaults to the value of ``create_graph``.
            create_graph (bool, optional): If ``True``, graph of the derivative will
                be constructed, allowing to compute higher order derivative products.
                Defaults to ``False``.
        """
        variables = (variables,) if isinstance(variables, Variable) else tuple(variables)

        if grad_variables is None:
            grad_variables = [None] * len(variables)
        elif isinstance(grad_variables, Variable) or torch.is_tensor(grad_variables):
            grad_variables = [grad_variables]
        else:
            grad_variables = list(grad_variables)

        grad_variables = _make_grads(variables, grad_variables)
        if retain_graph is None:
            retain_graph = create_graph

        Variable._execution_engine.run_backward(
>           variables, grad_variables, retain_graph, create_graph)
E       RuntimeError: VariableType::_standard_gamma_grad NYI

torch/autograd/__init__.py:83: RuntimeError
```
</details>

**Comment by fritzo** [comment]
Git bisect shows @gchanan's f5de5a84 as the offending commit.

**Comment by fritzo** [comment]
@gchanan It looks like you might be missing a `variants: function` line in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L105

**Comment by gchanan** [comment]
@fritzo Interesting that the bisect found that commit; that shouldn't have any effect, maybe a bug with gen_variable_type.  In any case https://github.com/pytorch/pytorch/pull/4327 should fix it -- sorry about that!

**Comment by fritzo** [comment]
Fixed by #4327
